lib/src/main/java/com/basistheory/android/service/CardBrandEnricher.kt

```diff
@@ -5,49 +5,38 @@ import com.basistheory.android.constants.CardBrands
 class CardBrandEnricher {
 
     object CardMasks {
-        const val MASK_4_8_12GAPS_19LENGTH = "#### #### #### #######"
-        const val MASK_4_8_12GAPS_16LENGTH = "#### #### #### ####"
-        const val MASK_4_10GAPS_15LENGTH = "#### ###### #####"
-        const val MASK_4_10GAPS_19LENGTH = "#### ###### #########"
+        const val MASK_4_8_12GAPS_19DIGITS = "#### #### #### #######"
+        const val MASK_4_8_12GAPS_16DIGITS = "#### #### #### ####"
+        const val MASK_4_10GAPS_15DIGITS = "#### ###### #####"
+        const val MASK_4_10GAPS_19DIGITS = "#### ###### #########"
     }
 
     object CvcMasks {
         const val THREE_DIGIT = "###"
         const val FOUR_DIGIT = "####"
     }
 
-    class CardDetails(
+    class CardBrandDetails(
         var brand: String = "",
         var identifierRanges: List<Pair<String, String?>>,
-        var validLengths: IntArray,
+        var validDigitCounts: IntArray,
+        var gapCount: Int,
```

Comment:
This was a bit misleading before - validLengths meant counts of valid numbers of digits, not including the gaps, so where this was used relied upon some implicit logic to strip out non-numeric characters first. This didn't work well when checking if the mask was satisfied from TextElement to set the new `isMaskSatisfied` flag. Changed this to track number of gaps before passing in valid lengths into the mask.